### PR TITLE
[Bug] Hotfix for Starter select UI with Filter

### PR DESCRIPTION
**Pull request by KimJeongSun**
## What are the changes?

This PR fixes the bug between the starter select UI with filter #2916 merged in beta and @Opaque02 's Starter UI selection update #1983 .

Most of the bugs were identified with the help of the previous PR contributor, @Opaque02

[bug discussion thread in discord](https://discord.com/channels/1125469663833370665/1265591426851409973)

## Why am I doing these changes?

The bugs in question range from mildly annoying ones to very dangerous issues, such as being able to select Pokémon that are not unlocked.

## What did change?

The bugs are as follows:
- [x] 1. When re-selecting the chosen Pokémon, the remove Pokémon menu does not appear.
- [x] 2. A bug in challenge mode where the "add to party" menu appears for invalid Pokémon.
- [x] 3. A bug where the starter Pokémon is removed from an incorrect position when removing Pokémon.
- [x] 4. When no starter Pokémon is selected, the cursor disappears when pressing the down button on the start button.
- [x] 5. A bug where the 'on' indicator for the ALL filter dropdown remains even when the entire dropdown is not activated.
- [x] 6. A bug where the starter cursor and Pokerus cursor appear outside the screen.

### Screenshots/Videos

plan to add videos and images
1. When re-selecting the chosen Pokémon, the remove Pokémon menu does not appear.
- before ( bug )
`remove from party` menu doesn't show.

https://github.com/user-attachments/assets/c435d5af-4f4b-459d-9b5f-7e45cf0ab539

- after ( fixed )
`remove from party` menu show and work properly

https://github.com/user-attachments/assets/8909b9e8-a542-4657-9767-a85d5b057685

2. A bug in challenge mode where the "add to party" menu appears for invalid Pokémon.
- before ( bug )

`add to party` menu show when select invalid pokemon

https://github.com/user-attachments/assets/285c2706-a16d-4c61-b20c-098edd383662

- after ( fixed )
`add to party` menu doesn't show when select invalid pokemon

https://github.com/user-attachments/assets/32211c13-9976-46dd-8e30-e7e57a648b13

3. A bug where the starter Pokémon is removed from an incorrect position when removing Pokémon.
- before ( bug )
When adding Pokémon in the reverse order displayed on the screen, the `starterCursor` is incorrectly removed during `remove from party`.

https://github.com/user-attachments/assets/f62b887c-3cde-4a19-a865-b34205f3f65d

- after ( fixed )
In all other cases, the removal happens correctly during `remove from party`.

https://github.com/user-attachments/assets/7fe11c16-de9e-4bde-8a13-8919b8096625

4. When no starter Pokémon is selected, the cursor disappears when pressing the down button on the start button.
- before ( bug )
If nothing is selected in the starter state and you press the down arrow key on the start button, the UI breaks.

https://github.com/user-attachments/assets/e77f75b0-1277-4ef2-8509-4b94cfb5d45a

- after ( fixed )
If nothing is selected in the starter state and you press the down arrow key on the start button, the cursor moves to the filter.

https://github.com/user-attachments/assets/65c416c1-063b-4a4d-bbe9-1b771ba28a9b

5. A bug where the 'on' indicator for the ALL filter dropdown remains even when the entire dropdown is not activated.
- before ( bug )
When some dropdowns are turned OFF in the filter, ALL remains ON.

https://github.com/user-attachments/assets/68e3d117-758f-4333-aa0c-7de738001e48

- after ( fixed )
When some dropdowns are turned OFF in the filter, ALL will also turn OFF.

https://github.com/user-attachments/assets/67a937b2-fa42-43eb-ba21-d481f8d31365

6. A bug where the starter cursor and Pokerus cursor appear outside the screen.
- before ( bug )
The pokerus cursor and starter cursor outside the screen are visible.

https://github.com/user-attachments/assets/7f39b153-9f4a-4a74-aef1-b9fbf96fde64

- after ( fixed )
The pokerus cursor and starter cursor outside the screen are no longer visible.

https://github.com/user-attachments/assets/81968ae5-b5e6-4796-af08-7239f4e015ba

## How to test the changes?

For the beta branch, all bugs can be reproduced.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by Madmadness65 — APPROVED** [review]
Thanks for the quick fixes. I'll hold off on merging it for a short while to let others review the changes too, to make sure nothing else unintentionally broke.

File: src/ui/starter-select-ui-handler.ts [hunk 1]
```diff
@@ -1150,75 +1154,69 @@ export default class StarterSelectUiHandler extends MessageUiHandler {
...
+          const currentPartyValue = this.starterSpecies.map(s => s.generation).reduce((total: number, gen: number, i: number) => total += this.scene.gameData.getSpeciesStarterValue(this.starterSpecies[i].speciesId), 0);
+          const newCost = this.scene.gameData.getSpeciesStarterValue(species.speciesId);
+          if (!isDupe && isValidForChallenge.value && currentPartyValue + newCost <= this.getValueLimit()) { // this checks to make sure the pokemon doesn't exist in your party, it's valid for the challenge and that it won't go over the cost limit; if it meets all these criteria it will add it to your party
+            options = [
+              {
+                label: i18next.t("starterSelectUiHandler:addToParty"),
+                handler: () => {
+                  ui.setMode(Mode.STARTER_SELECT);
+
+                  if (!isDupe && isValidForChallenge.value && this.tryUpdateValue(this.scene.gameData.getSpeciesStarterValue(species.speciesId), true)) {
```
Comment (flx-sta) [inline src/ui/starter-select-ui-handler.ts:1190]: NIT: Having a call to `tryUpdateValue` inside an if-condition is not the most beautiful and readable way to write it

File: src/ui/starter-select-ui-handler.ts [hunk 2]
```diff
@@ -2867,7 +2898,7 @@ export default class StarterSelectUiHandler extends MessageUiHandler {
     for (let s = 0; s < this.starterSpecies.length; s++) {
       const isValidForChallenge = new Utils.BooleanHolder(true);
       const species = this.starterSpecies[s];
-      Challenge.applyChallenges(this.scene.gameMode, Challenge.ChallengeType.STARTER_CHOICE, species, isValidForChallenge, this.scene.gameData.getSpeciesDexAttrProps(species, this.dexAttrCursor), !!this.starterSpecies.length);
+      Challenge.applyChallenges(this.scene.gameMode, Challenge.ChallengeType.STARTER_CHOICE, species, isValidForChallenge, this.scene.gameData.getSpeciesDexAttrProps(species, this.dexAttrCursor), !!(this.starterSpecies.length), false, false);
```
Comment (flx-sta) [inline src/ui/starter-select-ui-handler.ts:2901]: We don't check `.length` for it's boleanish value. As with any number we compare it to another number.
```suggestion
      Challenge.applyChallenges(this.scene.gameMode, Challenge.ChallengeType.STARTER_CHOICE, species, isValidForChallenge, this.scene.gameData.getSpeciesDexAttrProps(species, this.dexAttrCursor), this.starterSpecies.length !== 0, false, false);
```

**Comment by Tempo-anon** [comment]
Time to test in beta

**Comment by KimJeongSun** [comment]
@flx-sta Thank you for your comment. 
Since the hotfix PR has already been merged, I'll proceed with the update in a different PR. 
Please review it again then!

**Pull request merged**